python/ray/_private/serialization.py

```diff
@@ -423,8 +423,14 @@ def _serialize_to_msgpack(self, value):
         contained_object_refs = []
 
         if isinstance(value, RayTaskError):
-            metadata = str(ErrorType.Value("TASK_EXECUTION_EXCEPTION")).encode("ascii")
-            value = value.to_bytes()
+            if issubclass(value.cause.__class__, TaskCancelledError):
+                metadata = str(ErrorType.Value("TASK_CANCELLED")).encode("ascii")
```

Comment:
add a comment on why handling `TaskCancelledError` separately?
Also, why setting the metadata to `TASK_CANCELLED` can suppress the error? are we handling this metadata somewhere else?

Reply:
Yeah we only warn on TASK_EXCECUTION_EXCEPTION, not TASK_CANCELLED.

Reply:
Actually, I wonder we should just make ray.cancel never raise RayTaskError. It seems to be a bug that we raise RayTaskError when tasks are canceled (it is from https://github.com/ray-project/ray/blob/master/python/ray/exceptions.py#L147. I think because TaskCancelledError is a subclass of RayError, we happen to raise RayTaskError, not triggering `as_instanceof_cause` properly).